src/user/user.service.ts

```diff
@@ -42,20 +42,30 @@ export class UserService {
     }
 }
   async MyProfile(id: number): Promise<User | undefined> {
-    return this.userRepository.findOne({ where: { id } });
+    return this.userRepository.findOne({ where: { id, isDeleted: false } });
   }
   async findByEmail(email: string): Promise<User | undefined> {
     return this.userRepository.findOne({
-        where: { email },
+        where: { email, isDeleted: false },
         select: ['id', 'email', 'name', 'phone', 'country', 'birthday', 'role'],
     });
 }
   async findAll(): Promise<User[]> {
       return this.userRepository.find({
-          select: ['id', 'email', 'name', 'phone', 'country', 'birthday', 'role'],
+          where: { isDeleted: false },
+          select: ['id', 'email', 'name', 'phone', 'country', 'birthday', 'role', 'isDeleted', 'deletedDate'],
       });
 
   }
+  async softDelete(id: number): Promise<User | undefined> {
+    const user = await this.userRepository.findOne({ where: { id } });
+    if (!user) {  
+        return undefined;
+    }
+    user.isDeleted = true;
```

Comment:
Instead of doing this, you can simply do

```
await this.userRepository.softRemove(user);
```

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/27/commits/200b86b0c16713c23653e196873ac7c0f5ad3ba3

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/27/commits/5657f8ccd61312b517ada70345cd19d2158d162b